src/oldCode/components/messageListener.ts

```diff
@@ -52,16 +53,21 @@ const punishSpammersAndTrolls = async (message: Message): Promise<boolean> => {
 };
 
 export const messageListener = async (message: Message): Promise<void> => {
-  if (await punishSpammersAndTrolls(message)) {
+  if (!client.user) {
     return;
   }
 
-  if (!client.user) {
+  // Ignore all messages created by the bot itself
+  if (message.author.id === client.user.id) {
+    return;
+  }
+
+  if (await punishSpammersAndTrolls(message)) {
     return;
   }
 
-  // Ignore all messages created by the bot and dms from users
-  if (message.author.id != client.user.id && message.channel instanceof TextChannel) {
+  // Ignore DMs; include channels of type TextChannel (regular text channel) and NewsChannel (announcements channel)
+  if (!(message.channel instanceof DMChannel)) {
```

Comment:
isn't isTextChannel better than isTextOrNews?

Reply:
In discord.js v13, there would also be ThreadChannel, and I guess we want to include that? The comment could then be updated accordingly.